policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -126,17 +127,23 @@ impl kubert::index::IndexNamespacedResource<Service> for Index {
             .or_insert_with(|| Namespace {
                 service_routes: Default::default(),
                 namespace: Arc::new(ns),
-                services: Default::default(),
+                service_info: self.service_info.clone(),
             })
-            .update_service(service.name_unchecked(), service_info);
+            .update_service(service.name_unchecked(), &service_info);
+
+        self.service_info.write().insert(
+            ServiceRef {
+                name: service.name_unchecked(),
+                namespace: service.namespace().expect("Service must have Namespace"),
```

Comment:
nit, not a huge deal and maybe not worth the effort: we could probably reduce memory usage a bit by cloning the `Namespace`'s `Arc`ed copy of its own name into the `ServiceRef`, instead of storing the real string. feel free to ignore this, especially if it makes the code too complicated.